src/dbally_cli/text2sql.py

```diff
@@ -0,0 +1,133 @@
+import asyncio
+import importlib
+import os
+import sys
+
+import click
+import sqlalchemy
+
+from dbally.llms.base import LLM
+from dbally_codegen.autodiscovery import configure_text2sql_auto_discovery
+from dbally_codegen.generator import Text2SQLViewGenerator
+
+
+@click.command(help="Generate a Text2SQL view definition file.")
+@click.option(
+    "--file_path",
+    default="text2sql_view.py",
+    prompt="File path",
+    help="The path to the file where the view will be generated.",
+)
+@click.option(
+    "--db_url",
+    default="sqlite://",
+    prompt="Database connection string",
+    help="The database connection string.",
+)
+@click.option(
+    "--llm_object",
+    default="examples.rec:llm",
+    prompt="LLM object",
+    help="The path to the LLM object.",
+)
+@click.option(
+    "--llm_description",
+    is_flag=True,
+    prompt="Generate description using LLM object?",
+    help="Generate a description using the LLM object.",
+)
+@click.option(
+    "--similarity_index_factory",
+    default="examples.rec:index_builder",
+    prompt="Similarity index factory",
+    help="The path to the similarity index factory.",
+)
+def generate_text2sql_view(
+    file_path: str,
+    db_url: str,
+    llm_object: str,
+    llm_description: bool,
+    similarity_index_factory: str,
+) -> None:
+    """
+    Generate a Text2SQL view definition file.
+
+    Args:
+        file_path: The path to the file where the view will be generated.
+        db_url: The database connection string.
+        llm_object: The path to the LLM object.
+        llm_description: Generate a description using the LLM object.
+        similarity_index_factory: The path to the similarity index factory.
+    """
+    click.echo("Generating Text2SQL view...")
+
+    root, ext = os.path.splitext(file_path)
```

Comment:
better use pathlib it has been added in python 3.4

Reply:
A matter of preference, it does not change the functionality.